unified-runtime/source/loader/layers/sanitizer/sanitizer_common/sanitizer_utils.hpp

```diff
@@ -68,4 +69,38 @@ EnqueueUSMBlockingSet(ur_queue_handle_t Queue, void *Ptr, char Value,
 void PrintUrBuildLogIfError(ur_result_t Result, ur_program_handle_t Program,
                             ur_device_handle_t *Devices, size_t NumDevices);
 
+/**
+ * A RAII helper to run callback when destructing, useful for clean up resources
+ */
+struct ScopeGuard {
```

Comment:
This exists:
https://github.com/intel/llvm/blob/sycl/unified-runtime/source/common/ur_util.hpp#L509-L528